src/sentry/migrations/0020_hide_environment_none_20191126.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+def hide_environent_none(apps, schema_editor):
+    """
+    Hide environments that are named none, since they're blacklisted and no longer can be created.
+
+    We should iterate over each environment row individually in python instead so that we don't lock the DB up. This is
+    far slower but much safer
+    """
+    Environment = apps.get_model("sentry", "Environment")
+    EnvironmentProject = apps.get_model("sentry", "EnvironmentProject")
+    for env in Environment.objects.all():
```

Comment:
You need to use `RangedQuerySetWrapperWithProgress` or whatever it’s called. There should be many examples of this with prior art.

Reply:
The examples are now actually all gone, since they only existed in the new deleted `south_migrations` folder. Essentially you don't want to iterate over `.all()` here, since it'll bring all objects into memory at once. `RangeQuerySetWrapperWithProgressBar` iterates through all the rows in chunks.

If you want to see some examples, https://github.com/getsentry/sentry/commit/12aaaf29b4981673c13c1fcdb05d4bfbebf2258b has all south migrations, so you can take a look there.